src/subscriptions.js

```diff
@@ -0,0 +1,84 @@
+// Manages web push subscriptions
+//
+function urlBase64ToUint8Array(base64String) {
+  const padding = "=".repeat((4 - (base64String.length % 4)) % 4);
+  const base64 = (base64String + padding).replace(/-/g, "+").replace(/_/g, "/");
+
+  const rawData = window.atob(base64);
+  const outputArray = new Uint8Array(rawData.length);
+
+  for (let i = 0; i < rawData.length; ++i) {
+    outputArray[i] = rawData.charCodeAt(i);
+  }
+  return outputArray;
+}
+
+// In local storage: Map of sid -> { endpoint: "<endpoint>" }
+// If entry exists, it means there is a subscription to that room, wired to that endpoint.
+const LOCAL_STORE_KEY = "___hubs_subscriptions";
+
+export default class Subscriptions {
+  constructor(hubId) {
+    this.hubId = hubId;
+  }
+
+  setHubChannel = hubChannel => {
+    this.hubChannel = hubChannel;
+  };
+
+  setRegistration = registration => {
+    this.registration = registration;
+  };
+
+  setVapidPublicKey = vapidPublicKey => {
+    this.vapidPublicKey = vapidPublicKey;
+  };
+
+  getSubscriptionsFromStorage = () => {
```

Comment:
i tried adding a generic map to the JSON schema, and it complained -- and then on reflection it struck me that this stuff doesn't really seem to belong there anyway. i think the store schema should be pretty well defined sets of k/v pairs, instead of a grabbag of options which could lead to surprises.